test/specs/map/PreviewFeaturesService.spec.js

```diff
@@ -170,12 +177,9 @@ describe('ga_previewfeatures_service', function() {
           expect(layer).to.be.an(ol.layer.Vector);
           expect(layer.getSource().getFeatures().length).to.be(4);
           expect(spy.calledWith(map)).to.be(true);
-
-          // verify the request are stored in angular cache
-          gaPreviewFeatures.addBodFeatures(map, ids).then(function(feats) {
-            done();
-          });
-        });
+        }).then(done, done);
```

Comment:
you don't want that `done` was executed if the promise failed.

I suggest you leave as it was before, with the test to verify if the request is stored in the cache.